game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/retreat/OffensiveSubsRetreat.java

```diff
@@ -59,13 +66,36 @@ public Order getOrder() {
   public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
     if (battleState.isOver()
         || isDestroyerPresent()
-        || !isEvaderPresent()
-        || !isRetreatPossible()
+        || isEvaderNotPresent()
+        || isRetreatNotPossible()
         || isAutoWinScenario()) {
       return;
     }
-    battleActions.queryRetreat(
-        false, RetreatType.SUBS, bridge, battleState.getAttackerRetreatTerritories());
+
+    final Collection<Unit> unitsToRetreat =
+        CollectionUtils.getMatches(
+            battleState.getUnits(BattleState.Side.OFFENSE), Matches.unitCanEvade());
+    if (unitsToRetreat.isEmpty()) {
+      return;
+    }
+
+    final Collection<Territory> retreatTerritories;
+    if (Properties.getSubmersibleSubs(battleState.getGameData())) {
```

Comment:
nit, good place for a ternary here. Any thoughts or objections to that?

Reply:
With a ternary done, one could even consider inlining and you'd remove the intermediate variable as well.